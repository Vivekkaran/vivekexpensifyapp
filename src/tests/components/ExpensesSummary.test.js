import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<expenseSummary expenseCount={1} expenseTotal={235}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with multiple expenses', () => {
  const wrapper = shallow(<expenseSummary expenseCount={23} expenseTotal={2359098}/>);
  expect(wrapper).toMatchSnapshot();
});
