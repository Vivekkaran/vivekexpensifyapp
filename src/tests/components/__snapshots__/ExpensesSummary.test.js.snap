// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <expenseSummary
    expenseCount={23}
    expenseTotal={2359098}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "expenseCount": 23,
      "expenseTotal": 2359098,
    },
    "ref": null,
    "rendered": null,
    "type": "expenseSummary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "expenseCount": 23,
        "expenseTotal": 2359098,
      },
      "ref": null,
      "rendered": null,
      "type": "expenseSummary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should correctly render expenses summary with one expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <expenseSummary
    expenseCount={1}
    expenseTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "expenseCount": 1,
      "expenseTotal": 235,
    },
    "ref": null,
    "rendered": null,
    "type": "expenseSummary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "expenseCount": 1,
        "expenseTotal": 235,
      },
      "ref": null,
      "rendered": null,
      "type": "expenseSummary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
